### ISSUE TITLE
Export as svg / pdf?

### ISSUE THREAD

**Issue by skyhirider** [eligrey/FileSaver.js#176]
Can FileSaver save as .svg and .pdf ?

There isn't a svg example on the demo page but there are topics about svg here so I am unsure.

**Comment by tinaramsey87** [comment]
I've had success saving files as .pdf
Not sure about .svg

**Comment by jmcollin78** [comment]
@tinaramsey87 Can you give me a example for saving as pdf please ?
My app download a pdf after server generation and I try to force the browser to download the content.
Many thank's in advance.

**Comment by tinaramsey87** [comment]
Sorry it has taken me so long to reply!
First of all, your file needs to be in blob form for FileSaver to save it using its `saveAs()` function. My file was in base64 so I had to convert it to a blob - here's some code to do that http://stackoverflow.com/questions/16245767/creating-a-blob-from-a-base64-string-in-javascript *(use the first answer)* - it will look something like:
```base64toBlob: function(b64Data, contentType, sliceSize) {
		contentType = contentType || '';
		sliceSize = sliceSize || 512;

		var byteCharacters = atob(b64Data);
		var byteArrays = [];

		for (var offset = 0; offset < byteCharacters.length; offset += sliceSize) {
			var slice = byteCharacters.slice(offset, offset + sliceSize);

			var byteNumbers = new Array(slice.length);
			for (var i=0; i<slice.length; i++) {
				byteNumbers[i] = slice.charCodeAt(i);
			}

			var byteArray = new Uint8Array(byteNumbers);

			byteArrays.push(byteArray);
		}
		var blob = new Blob(byteArrays, {type: contentType});
		return blob;
	},```

fileData will be the base64 string (but you might already be working with a blob, so you may not need that step). contentType _must be_ 'application/pdf' to save as a pdf. Here's my code to save:
```var blob = base64toBlob(fileData, 'application/pdf')
saveAs(blob, fileName)```
the fileData and fileName were returned from the server.

**Comment by jmcollin78** [comment]
Many thank's for reply. I will check that rapidly

**Comment by tinaramsey87** [comment]
OK let me know if that works for you!

**Comment by migorman** [comment]
Hi @tinaramsey87 please i'm using FileSaver but it works for me just with .txt can u provide me your solution that works even for pfd or jpg ...   
 var blob = new Blob([data], { type: contentType });             
 FileSaver.saveAs(blob, fileName);

**Comment by tinaramsey87** [comment]
@migorman the code I posted should work. If you have code that you're trying to use post it and I can maybe help...

**Comment by jmcollin78** [comment]
Finally I gave up without success. I finally use the old good method: generate pdf server side and download it.

**Comment by migorman** [comment]
thanks @tinaramsey87  for your reply i'm just  doing like this :+1: 
 Svc.download(id).success(function(data, status, headers, config){
                var blob = new Blob([data], { type: contentType });
                FileSaver.saveAs(blob, fileName);
            })

**Comment by tinaramsey87** [comment]
Oh, @migorman , you need to declare contentType as 'application/pdf'. I don't see where you've declared it so make sure you're doing that somewhere. You can just do:
var blob = new Blob([data], {type: 'application/pdf'});

**Comment by migorman** [comment]
@tinaramsey87  yeah sure contentType was a variable in my casa and it toooks 'application/pdf and diffrent other values,i have found a solution thanks for your help :)

**Comment by tinaramsey87** [comment]
I am glad it worked for you :)

**Comment by jimmywarting** [comment]
- Content type shouldn't matter so much when saving the file. Content-type is only there to help browser render the file correctly it won't get baked into the file.
- `FileSaver.js` can save any format but don't expect it to convert the input to pdf, svg, txt, html, png, jpg or anything else. It's the content that you give to FileSaver that will be saved. 
- If you have a jpg image and need to save it as png then you have to convert the binary/blob first before giving it to the FileSaver

Bottom line is FileSaver don't have something called "Save as" it only has "Save blob"

Think we can close this.
